core/actor/src/main/scala/net/liftweb/actor/LAFuture.scala

```diff
@@ -333,32 +351,64 @@ object LAFuture {
     }
   }
 
-
   /**
-   * Collect all the future values into the aggregate future
-   * The returned future will be satisfied when all the
-   * collected futures are satisfied
+   * Given handlers for a value's success and failure and a set of futures, runs
+   * the futures simultaneously and invokes either success or failure callbacks
+   * as each future completes. When all futures are complete, if the handlers
+   * have not either satisfied or failed the overall result, `allValuesCompleted`
+   * is called to complete it. If it *still* isn't complete, the overall result
+   * is failed with an error.
+   *
+   * Note that the success and failure functions are guaranteed to be run in a
+   * thread-safe manner. Each is passed the value, the result future, the
+   * accumulating `ArrayBuffer`, and the index of the future that has been
+   * completed. For the failure handler, the value is the `Box` of the failure.
    */
-  def collect[T](future: LAFuture[T]*): LAFuture[List[T]] = {
-    val result = new LAFuture[List[T]]
-    if (future.isEmpty) {
-      result.satisfy(Nil)
+  def collect[T, A](
+    valueSucceeded: (T, LAFuture[A], ArrayBuffer[Box[T]], Int)=>Unit,
+    valueFailed: (Box[Nothing], LAFuture[A], ArrayBuffer[Box[T]], Int)=>Unit,
+    allValuesCompleted: (LAFuture[A], ArrayBuffer[Box[T]])=>Unit,
```

Comment:
These function signatures are complex enough to warrant a type alias.

Also I'm a bit confused by the fact that`valueFailed` has `Box[Nothing]` instead of just `Failure`. We'll always get a `Failure` in that position, right?